### PR TITLE
Add NIX_CONFIG env var for applying nix.conf overrides

### PR DESCRIPTION
**Pull request by kampka**
This PR offers an alternative approach of providing nix config settings via environment variables as [propose by edolstra](https://github.com/NixOS/nix/pull/4163#issuecomment-711902262) in #4163

### CODE REVIEW

**Comment by zimbatm** [comment]
Make sure to add some test in tests/config.sh and document it in doc/manual/command-ref/env-common.xml and doc/manual/command-ref/conf-file.xml

**Comment by kampka** [comment]
@zimbatm done

File: doc/manual/command-ref/conf-file.xml [hunk 1]
```diff
@@ -36,6 +36,15 @@ client assumes that the daemon has already loaded them.
   <literal>:</literal> will be loaded in reverse order.
 </para>
 
+<para>
```
Comment (edolstra) [inline doc/manual/command-ref/conf-file.xml:39]: This file should have been deleted, please change `doc/manual/src/command-ref/conf-file-prefix.md` instead.

File: doc/manual/src/command-ref/env-common.md [hunk 2]
```diff
@@ -86,6 +86,11 @@ Most Nix commands interpret the following environment variables:
     from (defaults to the XDG spec locations). The variable is treated
     as a list separated by the `:` token.
 
+  - `NIX_OPTIONS`
```
Comment (edolstra) [inline doc/manual/src/command-ref/env-common.md:89]: This line needs two trailing spaces to ensure correct formatting as a definition list.

File: doc/manual/src/command-ref/conf-file-prefix.md [hunk 3]
```diff
@@ -19,6 +19,9 @@ By default Nix reads settings from the following places:
     and `XDG_CONFIG_HOME`. If these are unset, it will look in
     `$HOME/.config/nix.conf`.
 
+  - If `NIX_OPTIONS` is set, it's contents is treated as the contents of
```
Comment (edolstra) [inline doc/manual/src/command-ref/conf-file-prefix.md:22]:
```suggestion
  - If `NIX_OPTIONS` is set, its contents is treated as the contents of
```

File: doc/manual/src/command-ref/env-common.md [hunk 4]
```diff
@@ -86,6 +86,11 @@ Most Nix commands interpret the following environment variables:
     from (defaults to the XDG spec locations). The variable is treated
     as a list separated by the `:` token.
 
+  - `NIX_OPTIONS`  
+    Applies settings from Nix configuration from the environment.
+    The content is treated as if it was read from a Nix configuration file.
+    Settings are separated by the `\n` token.
```
Comment (edolstra) [inline doc/manual/src/command-ref/env-common.md:92]:
```suggestion
    Settings are separated by the newline character.
```

Comment (edolstra) [reply doc/manual/src/command-ref/env-common.md:92]: Otherwise it suggests you can separate them by writing an escaped "\n".

**Comment by edolstra** [comment]
Can you rebase on master where conf-file.xml is gone?

File: doc/manual/src/command-ref/env-common.md (hunk 4 above)
Comment (kampka) [reply doc/manual/src/command-ref/env-common.md:92]: Good point.

**Comment by kampka** [comment]
> Can you rebase on master where conf-file.xml is gone?

Done

**Comment by kampka** [comment]
@edolstra looks like the `conf-file.xml` is still on master though.

**Comment by edolstra** [comment]
Sorry, I pushed to the wrong branch. It's fixed now (bdf2bcc989348fdcf223c9e2a383618454453eb2).

**Comment by kampka** [comment]
Thanks for the expedient reviews :+1:

**Comment by knedlsepp** [comment]
Could it be that something went wrong here with the docs? It mentions `NIX_OPTIONS` while the code uses `NIX_CONFIG`.

**Pull request merged**